jsettlers.logic/src/main/java/jsettlers/logic/buildings/trading/TradingBuilding.java

```diff
@@ -174,4 +220,47 @@ private void drawWaypointLine(boolean draw) {
 	public EPriority[] getSupportedPriorities() {
 		return SUPPORTED_PRIORITIES;
 	}
+
+	public boolean setDock(int[] position) {
+		if (this.type != EBuildingType.HARBOR) {
+			return false;
+		}
+		if (this.dockPosition != null) { // replace dock
+			this.grid.setDock(this.dockPosition, false, this.getPlayer());
+		}
+		this.dockPosition = position;
+		this.grid.setDock(position, true, this.getPlayer());
+		this.shipWayStart = new ShortPoint2D
```

Comment:
yes, but too complicated and not really readable:
new EDirection(this.dockPosition[2], this.dockPosition[3]).getNextHexPoint(new ShortPoint2D(this.dockPosition[0], this.dockPosition[1]), 5)